meerk40t/core/elements/shapes.py

```diff
@@ -253,21 +253,33 @@ def effect_remove(
         if len(data) == 0:
             return
         for node in data:
-            nparent = node.parent
-            if nparent.type.startswith("effect"):
-                was_emphasized = node.emphasized
-                node._parent = None  # Otherwise add_node will fail below
-                try:
-                    idx = nparent._children.index(node)
-                    if idx >= 0:
-                        nparent._children.pop(idx)
-                except IndexError:
-                    pass
-                nparent.parent.add_node(node)
-                if len(nparent.children) == 0:
-                    nparent.remove_node()
-                else:
-                    nparent.altered()
+            eparent = node.parent
+            nparent = eparent
+            while True:
```

Comment:
This repeated code is also probably a node function worth writing. `find_ancestor(criteria)` or something. Also `while nparent:` is  probably better than a `while True:`